### PR TITLE
uses the etcd observers provided by library-go

### PR DESCRIPTION
**Pull request by p0lyn0mial**

### CODE REVIEW

**Comment by p0lyn0mial** [comment]
/hold

requires https://github.com/openshift/library-go/pull/1211

**Comment by hexfusion** [comment]
/lgtm
thanks @p0lyn0mial

File: pkg/operator/configobservation/configobservercontroller/observe_config_controller.go [hunk 1]
```diff
@@ -87,7 +87,7 @@ func NewConfigObserver(
 				SecretLister_:                kubeInformersForNamespaces.InformersFor(operatorclient.TargetNamespace).Core().V1().Secrets().Lister(),
 				ConfigSecretLister_:          kubeInformersForNamespaces.InformersFor(operatorclient.GlobalUserSpecifiedConfigNamespace).Core().V1().Secrets().Lister(),
 				OpenshiftEtcdEndpointsLister: kubeInformersForNamespaces.InformersFor("openshift-etcd").Core().V1().Endpoints().Lister(),
-				ConfigmapLister:              kubeInformersForNamespaces.InformersFor("openshift-etcd").Core().V1().ConfigMaps().Lister(),
+				ConfigmapLister_:             kubeInformersForNamespaces.ConfigMapLister(),
```
Comment (sttts) [inline pkg/operator/configobservation/configobservercontroller/observe_config_controller.go:90]: won't this be cluster-wide?

Comment (p0lyn0mial) [reply pkg/operator/configobservation/configobservercontroller/observe_config_controller.go:90]: if you do `List` then you will get all config maps. 

however, our config observers only accept `configobserver.Lister`  which must be cast to the desired type. We cast it to `ConfigMapLister` and use it as follows `ConfigMapLister().ConfigMaps(EtcdEndpointNamespace).Get(etcdEndpointName)` to get scoped resources.

File: vendor/github.com/openshift/api/operator/v1/0000_90_cluster_csi_driver_01_config.crd.yaml [hunk 2]
```diff
@@ -50,6 +50,7 @@ spec:
                 - csi.kubevirt.io
                 - csi.shared-resources.openshift.io
                 - diskplugin.csi.alibabacloud.com
+                - vpc.block.csi.ibm.io
```
Comment (hexfusion) [inline vendor/github.com/openshift/api/operator/v1/0000_90_cluster_csi_driver_01_config.crd.yaml:53]: how is this related?

Comment (hexfusion) [reply vendor/github.com/openshift/api/operator/v1/0000_90_cluster_csi_driver_01_config.crd.yaml:53]: I see just from the bump disregard.
